docs/docs-content/integrations/k3s.md

```diff
@@ -0,0 +1,148 @@
+logoUrl: 'https://registry.spectrocloud.com/v1/edge-k3s/blobs/sha256:5b7f8a80330d6938056848905238efbbd085e2f0f86fd03d7be10ccdabf791c7?type=image/png'
+tags: ["packs", "kubernetes", "k3s", "edge"]
+---
+
+K3s is a lightweight distribution of Kubernetes, specifically designed for resource-constrained environments and IoT applications. Unlike standard Kubernetes with multiple binaries for its various components, K3s is packaged as a single binary with a lightweight storage backend and minimal external dependencies. For more information about K3s, refer to [K3s documentation](https://docs.k3s.io/). 
+
+## Versions Supported
+
+<Tabs queryString="versions">
+<TabItem label="1.27.X" value="k3s_1.27">
+
+### Prerequisites
+
+- An edge device with AMD64(x86_64) processor architecture or a Palette Virtual Cluster. 
+- A minimum of 2 CPU cores and 1 GB memory. 
+
+### Parameters
+
+Since you can deploy both Palette Virtual Clusters and Edge clusters using K3s, you have different configuration options depending on the type of cluster.
+
+<Tabs queryString="cluster-types">
+
+<TabItem label="Edge" value="edge">
+
+|**Parameter**|**Description** |
+|-------------|----------------|
+| `cluster.config.cluster-cidr`| Specifies the CIDR range that can be used by pods in the cluster. | 
+| `cluster.config.service-cidr`| Specifies the CIDR range that can be used by services in the cluster.|
+| `kube-apiserver-arg`| This parameter contains extra arguments for the Kubernetes API server, such as enabling audit logging, enabling certain authorization modes, and setting profiling and secure-port.|
+|`kube-controller-manager-arg` | This parameter describes extra arguments for the Kubernetes Controller Manager, such as enabling certain feature gates and setting profiling. |
+| `kubelet-arg` |  This parameter contains extra arguments for Kubelet during node registration, such as setting feature gates, protecting kernel defaults, and disabling the read-only port. |
+| `pack.palette.config.oidc.identityProvider` | Dynamically enabled OpenID Connect (OIDC) Identity Provider (IDP) setting based on your UI selection when you add the K3s pack to your profile. This parameter appears in the YAML file after you make a selection. Refer to [Configure OIDC Identity Provider](#configure-custom-oidc). |
+
+You can add cloud-init stages, which allow you to customize your instances declaratively. The cloud-init stages are exposed by [Kairos](https://kairos.io/docs/architecture/cloud-init/), an open source project. For more information, check out the [Cloud Init Stages](../clusters/edge/edge-configuration/cloud-init.md) reference.
+</TabItem>
+
+<TabItem label="Palette Virtual Cluster" value="palette-virtual-cluster">
+
+Since you are setting up a virtual cluster inside another Kubernetes cluster, you can configure its pods and services differently than the host cluster. The default Kubeadm configuration file you get includes parameters that offer you a higher degree of customization.
```

Comment:
```suggestion
Since you are setting up a virtual cluster inside another Kubernetes cluster, you can configure its pods and services differently than the host cluster. The default kubeadm configuration file you get includes parameters that offer you a higher degree of customization.
```